test/dynamic-import-tests.js

```diff
@@ -0,0 +1,61 @@
+const assert = require("assert");
+
+describe("dynamic import", () => {
+  it("should support a standalone import() call", () => {
+    let count = 0;
+    const moduleImport = module.import;
+
+    module.import = function (id) {
+      count++;
+      return moduleImport.call(this, id);
+    };
+
+    import("./misc/abc");
+    module.import = moduleImport;
+    assert.strictEqual(count, 1);
+  });
+
+  it("should resolve as a namespace import", (done) => {
+    import("./misc/abc").then((ns) => {
+      assert.deepEqual(ns, { a: "a", b: "b", c: "c" });
+      done();
+    });
+  });
+
+  it("should support a variable id", (done) => {
+    const id = "./misc/abc";
+    import(id).then((ns) => {
+      assert.deepEqual(ns, { a: "a", b: "b", c: "c" });
+      done();
+    });
+  });
+
+  it("should support a template string id", (done) => {
+    const id = "./misc/abc";
+    import(`${id}`).then((ns) => {
+      assert.deepEqual(ns, { a: "a", b: "b", c: "c" });
+      done();
+    });
+  });
+
+  it("should support import() in an assignment", () => {
+    const p = import("./misc/abc");
+    assert.ok(p instanceof Promise);
+  });
+
+  it("should support import() in a function", () => {
+    function p() {
+      return import("./misc/abc");
+    }
+
+    assert.ok(p() instanceof Promise);
+  });
+
+  it("should support import() with yeild", () => {
+    function* p() {
+      yield import("./misc/abc");
+    }
+
+    assert.ok(p());
+  });
```

Comment:
Can we add a test that the properties of the namespace object get updated when the exported values change?